datasets/naip/naip.py

```diff
@@ -0,0 +1,159 @@
+            raise Exception(f"{cog_path} does not exist in {cog_storage}")
+
+        m = re.match(naip_regex, cog_path)
+        if not m:
+            raise Exception(f"{cog_path} did not match regex")
+
+        if YearMismatchErrorRecord.is_wrong_year(cog_path):
+            raise ValueError(f"Mismatched year for asset at path {cog_path}")
+
+        state = m.group("state")
+        year = m.group("year")
+        image_dir = m.group("image_dir")
+        area = m.group("area")
+        fname = m.group("fname")
+
+        # Remove any processing numbers from the fname
+        # for FGDC
+        fgdc_fname = "_".join(fname.split("_")[:6])
+
+        fgdc_path = (
+            f"v002/{state}/{year}/{state}_fgdc_{year}" f"/{area}/{fgdc_fname}.txt"
+        )
+
+        thumbnail_name = f"{fname}.200.jpg"
+        thumbnail_path = f"v002/{state}/{year}/{image_dir}" f"/{area}/{thumbnail_name}"
+
+        # Account for the fact that some images don't have FGDC metadata
+        if not cog_storage.file_exists(fgdc_path):
+            fgdc_exists = False
+        else:
+            fgdc_exists = True
+
+        fgdc_href = cog_storage.get_url(fgdc_path)
+        cog_href = cog_storage.get_url(cog_path)
+
+        if not cog_storage.file_exists(thumbnail_path):
+            logger.debug("Creating thumbnail %s", thumbnail_path)
+            thumbnail_f = functools.partial(create_thumbnail, cog_href, thumbnail_name)
+            with_backoff(thumbnail_f, is_throttle=lambda x: True)
+            kwargs = {}
+            if isinstance(cog_storage, BlobStorage):
+                kwargs["content_settings"] = azure.storage.blob.ContentSettings(
+                    content_type="image/jpeg"
+                )
+
+            cog_storage.upload_file(thumbnail_name, thumbnail_path, **kwargs)
+
+        assert cog_storage.file_exists(thumbnail_path)
+
+        thumbnail_href = cog_storage.get_url(thumbnail_path)
+
+        logger.debug("Creating STAC Item for {}...".format(cog_href))
+        try:
+            item = stac.create_item(
+                state=state,
+                year=year,
+                fgdc_metadata_href=fgdc_href if fgdc_exists else None,
+                cog_href=cog_href,
+                thumbnail_href=thumbnail_href,
+            )
+        except UnicodeDecodeError:
+            # Try once more, but without fgdc
+            item = stac.create_item(
+                state=state,
+                year=year,
+                fgdc_metadata_href=None,
+                cog_href=cog_href,
+                thumbnail_href=thumbnail_href,
+            )
+
+        # Compatibility with older STAC items.
+        item.properties.pop("grid:code", None)
+        item.assets["image"].extra_fields.pop("raster:bands", None)
+        item.stac_extensions = [
+            "https://stac-extensions.github.io/eo/v1.0.0/schema.json",
+            "https://stac-extensions.github.io/projection/v1.0.0/schema.json",
+        ]
```

Comment:
These changes are all related to a bump from an old commit of stactooks-naip (in the 0.1.x range IIRC) to 0.3.1. The older image seemed to have some difficulty in the environment expected by pctasks, though I don't recall the exact errors.

With these changes, we can build an item using the newer stactools, but produce an identical item to the old.